x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -3546,6 +3567,34 @@ export class RulesClient {
         })
       );
     }
+
+    // check for actions using frequency params if the rule has global frequency params defined
+    if (usesGlobalFreqParams) {
+      const actionsWithFrequency = actions.filter((action) => Boolean(action.frequency));
+      if (actionsWithFrequency.length) {
+        throw Boom.badRequest(
+          i18n.translate('xpack.alerting.rulesClient.validateActions.mixAndMatchFreqParams', {
+            defaultMessage:
+              'Cannot mix and match per-action frequency params when notify_when or throttle are globally defined: {groups}',
```

Comment:
"Cannot specify per-action frequency params when notify_when and throttle are defined at the rule level"